sfge/src/main/java/com/salesforce/rules/fls/apex/AbstractFlsVisitor.java

```diff
@@ -173,4 +147,70 @@ private void recursivelyProcessConditionChildren(
             }
         }
     }
+
+    /**
+     * Indicates whether a boolean expression should be processed recursively, based on the nature
+     * of its operator, its children, and whether its satisfaction is expected for this path.
+     */
+    // TODO: This implementation is somewhat simplistic, and there are definitely
+    //       cases that it doesn't support. These cases will lead to false positives.
+    //       If unsupported edge cases crop up frequently in the wild, we should
+    //       enhance this to be more intelligent.
+    private boolean booleanExprRequiresRecursion(
```

Comment:
Can we have the methods related to boolean expression processing in a separate util class? A lot of this looks reusable in other contexts too.